File: day1/golf.py
r = "1000\n1000\n20\n1020" #Some input with numbers separated by newlines

from itertools import combinations 
from math import prod
s = lambda r,n: prod([x for x in combinations([int(k) for k in r.splitlines()],n) if sum(x)==2020][0])
part1 = s(r,2)
part2 = s(r,3)

File: day1/golf.js
const p=r=>r.split("\n").map(e=>+e)

const part1=(r,o=0)=>((a=p(r)).find((e,i)=>a.find((k,j)=>j!=i&&(e+k)==2020&&(o=e*k))),o)

const part2=(r,o=0)=>((a=p(r)).find((e,i)=>a.find((k,j)=>a.find((n,g)=>g!=i&&j!=i&&g!=j &&(e+k+n)==2020&&(o=e*k*n)))),o)
